azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackRequestMaker.java

```diff
@@ -0,0 +1,235 @@
+      throw new NullPointerException("props argument can't be null");
+    }
+    instance = new JobCallbackRequestMaker(props);
+    isInitialized = true;
+    logger.info("Initialization for " + JobCallbackRequestMaker.class.getName()
+        + " is completed");
+  }
+
+  public static boolean isInitialized() {
+    return isInitialized;
+  }
+
+  public static JobCallbackRequestMaker getInstance() {
+    if (!isInitialized) {
+      throw new IllegalStateException(JobCallbackRequestMaker.class.getName()
+          + " hasn't initialzied");
+    }
+    return instance;
+  }
+
+  private JobCallbackRequestMaker(Props props) {
+
+    int connectionRequestTimeout =
+        props.getInt("jobcallback.connection.request.timeout",
+            DEFAULT_TIME_OUT_MS);
+
+    int connectionTimeout =
+        props.getInt("jobcallback.connection.timeout", DEFAULT_TIME_OUT_MS);
+
+    int socketTimeout =
+        props.getInt("jobcallback.socket.timeout", DEFAULT_TIME_OUT_MS);
+
+    responseWaitTimeoutMS =
+        props.getInt("jobcallback.response.wait.timeout",
+            DEFAULT_RESPONSE_WAIT_TIME_OUT_MS);
+
+    logger.info("responseWaitTimeoutMS: " + responseWaitTimeoutMS);
+
+    RequestConfig requestConfig =
+        RequestConfig.custom()
+            .setConnectionRequestTimeout(connectionRequestTimeout)
+            .setConnectTimeout(connectionTimeout)
+            .setSocketTimeout(socketTimeout).build();
+
+    logger.info("Global request configuration " + requestConfig.toString());
+
+    HttpClient httpClient =
+        HttpClientBuilder.create().setDefaultRequestConfig(requestConfig)
+            .build();
+
+    int jobCallbackThreadPoolSize =
+        props.getInt("jobcallback.thread.pool.size", DEFAULT_THREAD_POOL_SIZE);
+    logger.info("Jobcall thread pool size: " + jobCallbackThreadPoolSize);
+
+    ExecutorService executorService =
+        Executors.newFixedThreadPool(jobCallbackThreadPoolSize);
+    futureRequestExecutionService =
+        new FutureRequestExecutionService(httpClient, executorService);
+  }
+
+  public FutureRequestExecutionMetrics getJobcallbackMetrics() {
+    return futureRequestExecutionService.metrics();
+  }
+
+  public void makeHttpRequest(String jobId, Logger logger,
+      List<HttpRequestBase> httpRequestList) {
+
+    if (httpRequestList == null || httpRequestList.isEmpty()) {
+      logger.info("No HTTP requests to make");
+      return;
+    }
+
+    for (HttpRequestBase httpRequest : httpRequestList) {
+
+      logger.info("Job callback http request: " + httpRequest.toString());
+      logger.info("headers [");
+      for (Header header : httpRequest.getAllHeaders()) {
+        logger.info(String.format("  %s : %s", header.getName(),
+            header.getValue()));
+      }
+      logger.info("]");
+
+      HttpRequestFutureTask<Integer> task =
+          futureRequestExecutionService.execute(httpRequest,
+              HttpClientContext.create(), new LoggingResonseHandler(logger));
```

Comment:
It just takes in the logger instance in JobCallbackRequestMaker, right? In this case, you can even make LoggingResonseHandler a member class, then you don't have to pass in the logger.